### PR TITLE
[PP] Bypass seed checkpoint my init-ing model parts separately

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #473
* #517
* __->__ #516

Allows PP to be used without a seed checkpoint by calling `init_weight` on each model part. This is the solution in step 1 of https://github.com/pytorch/torchtitan/issues/514 proposed by @wconstab

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -208,9 +206,10 @@ def loss_fn(pred, labels):
     checkpoint_loaded = checkpoint.load()
 
     if parallel_dims.pp_enabled and not checkpoint_loaded:
-        raise RuntimeError(
-            "Pipeline Parallelism requires meta-initialization and loading seed checkpoint. "
-            "Please run `./create_seed_checkpoint.sh` and rerun training with `--checkpoint.enable_checkpoint`"
+        # TODO: fix this by allowing each rank to set their own seed
```
Comment (wconstab) [inline train.py:207]: iiuc, init_weights would be called all the time, and then when seed ckpt is specified, it would overwrite the init.  that sgtm.

**Review by wconstab — APPROVED** [review]

File: train.py (hunk 1 above)
Comment (H-Huang) [reply train.py:207]: yep, that is correct
